src/utils/refreshDockerode.ts

```diff
@@ -72,14 +79,35 @@ async function getDockerodeOptions(newEnv: NodeJS.ProcessEnv): Promise<DockerOpt
     return undefined;
 }
 
-async function getSshAuthSock(): Promise<string | undefined> {
-    if (isWindows()) {
-        return '\\\\.\\pipe\\openssh-ssh-agent';
-    } else {
+async function validateSshAuthSock(newEnv: NodeJS.ProcessEnv): Promise<boolean> {
+    if (!newEnv.SSH_AUTH_SOCK && isWindows()) {
+        // On Windows, we can use this one by default
+        newEnv.SSH_AUTH_SOCK = '\\\\.\\pipe\\openssh-ssh-agent';
+    } else if (!newEnv.SSH_AUTH_SOCK) {
         // On Mac and Linux, if SSH_AUTH_SOCK isn't set there's nothing we can do
         // Running ssh-agent would yield a new agent that doesn't have the needed keys
-        await ext.ui.showWarningMessage('In order to use an SSH DOCKER_HOST on OS X and Linux, you must configure an ssh-agent.');
+        return false;
     }
+
+    const authSock = new Socket();
+    const cts = new CancellationTokenSource();

```

Comment:
A `CancellationTokenSource` is disposable, so should be disposed before the function returns, either in the `finally()` handler or you could add `try`/`finally` blocks.